### PR TITLE
feat(communities): Adds mute community intervals

### PR DESCRIPTION
**Pull request by borismelnik**
Fixes: #9369

### What does the PR do

Now community can be muted for some intervals

### Affected areas

Communities

### StatusQ checklist

- [ ] add documentation if necessary (new component, new feature)
- [ ] update sandbox app
  - in case of new component, add new component page
  - in case of new features, add variation to existing component page
  - nice to have: add it to the demo application as well
- [ ] test changes in both light and dark theme?

### Screenshot of functionality (including design for comparison)

- [ ] I've checked the design and this PR matches it

Снимок экрана 2023-06-07 в 13 43 40

Снимок экрана 2023-06-07 в 13 43 49

### CODE REVIEW

**Review by jrainville — APPROVED** [review]
Nice work

File: ui/app/mainui/AppMain.qml [hunk 1]
```diff
@@ -354,11 +355,21 @@ Item {
...
+
                         StatusAction {
-                            text: model.muted ? qsTr("Unmute Community") : qsTr("Mute Community")
-                            icon.name: model.muted ? "notification-muted" : "notification"
+                            enabled: model.muted
+                            text: qsTr("Unmute Community")
+                            icon.name: "notification-muted"
                             onTriggered: {
-                                communityContextMenu.chatCommunitySectionModule.setCommunityMuted(!model.muted)
+                                communityContextMenu.chatCommunitySectionModule.setCommunityMuted(Constants.MutingVariations.Unmuted)
```
Comment (caybro) [inline ui/app/mainui/AppMain.qml:372]: Nice but these mute/unmute menu items (perhaps the whole context menu?) just seems duplicated here and CommunitiesListPanel.qml. Maybe extract it to some share component?

Comment (borismelnik) [reply ui/app/mainui/AppMain.qml:372]: Yeah I thought about it, but there may be another list of menu items.

File: src/app_service/service/community/service.nim [hunk 2]
```diff
@@ -1695,10 +1695,11 @@ QtObject:
...
       error "Error banning user from community", msg = e.msg
 
-  proc setCommunityMuted*(self: Service, communityId: string, muted: bool) =
+  proc setCommunityMuted*(self: Service, communityId: string, mutedType: int) =
     try:
-      discard status_go.setCommunityMuted(communityId, muted)
-
+      discard status_go.setCommunityMuted(communityId, mutedType)
+      
+      let muted = if (MutedType(mutedType) == MutedType.Unmuted): false else: true
```
Comment (MishkaRogachev) [inline src/app_service/service/community/service.nim:1702]: If we don't relay on actual value from status-go, it's better to handle an error:
```
      let response = status_go.setCommunityMuted(communityId, mutedType)
      if not response.error.isNil:
        error "error muting the community", msg = response.error.message
        return
```

**Review by MishkaRogachev — APPROVED** [review]
Look good, just one question

**Comment by lukaszso** [comment]
@borismelnik QA results:
1. https://github.com/status-im/status-desktop/issues/11041 - reported separately by mistake but it was found during testing of this PR
2. The mute icon next to the chat/channel/category is misplaced and not fully visible
mute-1

**Comment by caybro** [comment]
> > The mute icon next to the chat/channel/category is misplaced and not fully visible
> 
> This one was also found by Igor and Lukas and prossibly already fixed.

Yup, already fixed here: https://github.com/status-im/status-desktop/pull/11053

**Review by lukaszso — APPROVED** [review]
QA Pass. Tested the functionality with chats, communities, categories and channels. Outside of the known issues there is nothing to report.

**Comment by caybro** [comment]
@jrainville looks we have a failing community related `e2e` testcase in master; it fails here and in my PR here too (https://github.com/status-im/status-desktop/pull/11066), both PRs seem unrelated to the failure

**Comment by borismelnik** [comment]
> @jrainville looks like we have a failing community related `e2e` testcase in master; it fails here and in my PR here too (#11066), both PRs seem unrelated to the failure

I will check today cos I change some UI @caybro

**Comment by caybro** [comment]
> > @jrainville looks like we have a failing community related `e2e` testcase in master; it fails here and in my PR here too (#11066), both PRs seem unrelated to the failure
> 
> I will check today cos I change some UI @caybro

But I didn't in my PR :) And they both have the same failure

**Comment by jrainville** [comment]
I think the errors are legit. You both touch the communities in the left bar. I tested master and the test works fine locally, but this PR failed for me locally.

**Comment by borismelnik** [comment]
> > I think the errors are legit. You both touch the communities in the left bar. I tested master and the test works fine locally, but this PR failed for me locally.
> 
> Too bad :/ will have a look or @borismelnik if you are faster :) The strange thing is that I don't deal with creating a community at all

will do it right now :)

**Comment by borismelnik** [comment]
working on it :(

**Review by caybro — APPROVED** [review]
Just the 2 minor things, otherwise looks nice!

File: ui/app/AppLayouts/Profile/panels/CommunitiesListPanel.qml [hunk 3]
```diff
@@ -9,15 +9,17 @@ import StatusQ.Popups 0.1
...
 StatusListView {
     id: root
 
     property bool hasAddedContacts: false
 
     signal inviteFriends(var communityData)
     signal closeCommunityClicked(string communityId)
-    signal leaveCommunityClicked(string community, string communityId, string outroMessage)
-    signal setCommunityMutedClicked(string communityId, bool muted)
+    signal leaveCommunityClicked(string community, string communityId, string outroMessage)
```
[the remark this answers is not in the document]
Comment (borismelnik) [reply ui/app/AppLayouts/Profile/panels/CommunitiesListPanel.qml]: done

**Pull request merged**